packages/peregrine/lib/talons/Navigation/useNavigation.js

```diff
@@ -11,16 +12,27 @@ const ancestors = {
     MENU: null
 };
 
-export const useNavigation = () => {
+export const useNavigation = props => {
+    const { customerQuery } = props;
+
     // retrieve app state from context
     const [appState, { closeDrawer }] = useAppContext();
     const [catalogState, { actions: catalogActions }] = useCatalogContext();
-    const [, { getUserDetails }] = useUserContext();
+    const [{ isSignedIn }, { actions: userActions }] = useUserContext();
+    const [getUserDetails, { data: getCustomerData }] = useLazyQuery(
+        customerQuery,
+        { fetchPolicy: 'network-only' }
+    );
 
-    // request data from server
     useEffect(() => {
-        getUserDetails();
-    }, [getUserDetails]);
+        if (isSignedIn) {
+            getUserDetails();
```

Comment:
Should you not `await` for this or use the output? Does that function cause any side effects like setting the `store` or `context`?

Reply:
I really wish `useLazyQuery` returned a Promise like `useMutation`, but it does not ☹️. This call does cause the effect to run again though, by updating `getCustomerData`. Is that okay?

Reply:
Maybe this would be a good candidate for the idea @jimbo had. Create a `useCustomerData` hook that returns an async callback (and uses refs internally to ensure we don't double fetch?). It can handle storing that data too Then you would just invoke that here.